File: kotlin/Math and Geometry/JosephusOptimized.kt
fun josephusOptimized(n: Int, k: Int): Int {
    var res = 0
    for (i in 2..n) {
        // res[i] = (res[i - 1] + k) % i.
        res = (res + k) % i
    }
    return res
}

File: cpp/Math and Geometry/josephus_optimized.cpp
int josephusOptimized(int n, int k) {
    int res = 0;
    for (int i = 2; i <= n; i++) {
        // res[i] = (res[i - 1] + k) % i;
        res = (res + k) % i;
    }
    return res;
}